### DISCUSSION TITLE
Azure OpenAI showing 404

### DISCUSSION THREAD

**Discussion by Uralstech** [RageAgainstThePixel/com.openai.unity#269] [Q&A]
# Bug Report

## Overview
When I'm using the threads API through Azure OpenAI, it shows a 404 error. I've been using this plugin for my Unity projects through the normal OpenAI API without any problems. I've created a model deployment in Azure OpenAI and went through the setup process for [authenticating the plugin for Azure](https://github.com/RageAgainstThePixel/com.openai.unity/?tab=readme-ov-file#azure-openai).

This is the error I am getting:
```
RestException: <b>[DELETE:404]</b> <color="cyan">https://xxxxx.openai.azure.com/openai/deployments/xxxxx/threads/xxxxx?api-version=2024-05-01-preview</color> <color="red">Failed!</color>
```

I believe the issue here is the `deployments/xxxxx/` part, as when I look at the code sample in the Azure OpenAI Studio, it shows a URL like so:
```
https://${endpoint}/openai/threads?api-version=2024-05-01-preview
```

I tried using the CURL sample from Azure OpenAI to create and delete threads, and it worked.

## Temporary Solution

In [OpenAIAuthSettingsInfo.cs](https://github.com/RageAgainstThePixel/com.openai.unity/blob/main/OpenAI/Packages/com.openai.unity/Runtime/Authentication/OpenAISettingsInfo.cs#L8), line 97, I changed:
```csharp
BaseRequest = $"/openai/deployments/{DeploymentId}/";
```
to
```csharp
BaseRequest = $"/openai/";
```
and it is working perfectly.

**Comment by StephenHodgson** [comment]
@Uralstech Did Azure change their url for deployments?

**Reply by Uralstech** [reply]
I don't know about that, but it starts working when I change `BaseRequest`.

**Reply by StephenHodgson** [reply]
Hmm doing some reading up on this. It seems they did change their endpoints for this preview version. I'll update it to support this soon.

**Reply by StephenHodgson** [reply]
yeah so GA it is fine, but you're using preview API. 

- https://learn.microsoft.com/en-us/azure/ai-services/openai/reference
- https://learn.microsoft.com/en-us/azure/ai-services/openai/reference-preview

**Reply by StephenHodgson** [reply]
I'll open a chore ticket in the main repo to update this

**Reply by StephenHodgson** [reply]
Opened a new ticket for tracking https://github.com/RageAgainstThePixel/OpenAI-DotNet/issues/342

**Reply by Uralstech** [reply]
I see. The Assistants API on Azure is in preview, and I didn't check the Completions endpoint, so I didn't know there was a difference in the link 😅

**Comment by StephenHodgson** [comment]
you could also workaround this by specifying custom domain:
```
https://YOUR_RESOURCE_NAME.openai.azure.com/openai/threads/{thread_id}/runs?api-version=2024-05-01-preview
```

**Reply by Uralstech** [reply]
Can I do this through the `ScriptableObject`? Or do I have to do this through code?

**Reply by StephenHodgson** [reply]
yes, that is correct. in OpenAIConfiguration or in the code. Up to you.

**Reply by Uralstech** [reply]
Thanks! The custom domain is the "Proxy Domain" field right? When I enable "Use Azure OpenAI", the field disappears, and I can't set it.

**Reply by StephenHodgson** [reply]
Ah my bad. Well in the meantime keep what you have and I'll have it fixed as soon as possible.

**Comment by Uralstech** [comment]
I should also add that Vision is not supported in assistant messages on Azure, although this is not related to the issue.
